service/debugger/debugger.go

```diff
@@ -168,16 +178,66 @@ func (d *Debugger) CreateBreakpoint(requestedBp *api.Breakpoint) (*api.Breakpoin
 	if err != nil {
 		return nil, err
 	}
-	bp.Tracepoint = requestedBp.Tracepoint
-	bp.Goroutine = requestedBp.Goroutine
-	bp.Stacktrace = requestedBp.Stacktrace
-	bp.Variables = requestedBp.Variables
-	bp.Cond = nil
+	if err := copyBreakpointInfo(bp, requestedBp); err != nil {
+		d.process.ClearBreakpoint(bp.Addr)
+		return nil, err
+	}
 	createdBp = api.ConvertBreakpoint(bp)
 	log.Printf("created breakpoint: %#v", createdBp)
 	return createdBp, nil
 }
 
+func (d *Debugger) AmendBreakpoint(amend *api.Breakpoint) error {
+	var original *proc.Breakpoint
+	for i := range d.process.Breakpoints {
+		if d.process.Breakpoints[i].ID == amend.ID {
+			original = d.process.Breakpoints[i]
+			break
+		}
+	}
+	if original == nil {
+		return fmt.Errorf("no breakpoint with ID %d", amend.ID)
+	}
+	if err := d.validBreakpointName(amend.Name, false); err != nil {
+		return err
+	}
+	copyBreakpointInfo(original, amend)
```

Comment:
Need to check the error here, especially since a condition is added via amend and could error out.